apps/api/src/app/messaging/kafka/kafka.service.ts

```
@@ -0,0 +1,182 @@
+          console.log('Retried in', delay, 'ms...');
+        }
+      }
+    }
+
+    throw new Error(`Failed to connect to Kafka after ${this.maxRetries} attempts`);
+  }
+
+  @Interval(process.env.KAFKA_HEALTH_CHECK_INTERVAL ? parseInt(process.env.KAFKA_HEALTH_CHECK_INTERVAL, 10) : 30000)
+  async healthCheck() {
+    if (!this.isConnected) return;
+
+    try {
+      await this.consumer.describeGroup();
+    } catch (error) {
+      this.logger.error(
+        '🔴 Kafka health check failed, attempting reconnection:',
+        (error instanceof Error ? error.message : String(error))
+      );
+      this.isConnected = false;
+
+      try {
+        await this.reconnect();
+      } catch (reconnectError) {
+        this.logger.error(
+          '💀 Reconnection failed:',
+           (reconnectError instanceof Error ? reconnectError.message : String(reconnectError))
+        );
+      }
+    }
+  }
+
+  private async reconnect(): Promise<void> {
+    await this.consumer.disconnect();
+    await this.connectWithRetry();
+    await this.subscribe();
+    await this.startConsumer();
+  }
+
+  private async startConsumer(): Promise<void> {
+    this.logger.log('🎧 Starting message listening...');
+
+    await this.consumer.run({
+      autoCommit: false,
+      eachMessage: async ({ topic, partition, message }: EachMessagePayload) => {
+        if (!this.isConnected) {
+          this.logger.warn('⚠️ Skipping message processing - Kafka disconnected');
+          return;
+        }
+
+        try {
+          await this.messageHandler.handleMessage(
+            message,
+            async () => {
+              await this.consumer.commitOffsets([{
+                topic,
+                partition,
+                offset: (parseInt(message.offset) + 1).toString()
+              }]);
+            }
+          );
+        } catch (error) {
+          this.logger.error(`💥 Error processing message ${message.offset}:`, error);
```

The error is logged but silently swallowed. If a handler fails, the message will never be reprocessed since you are using `autoCommit: false` but are not managing the retry logic.

Impact : Potential loss of critical messages (Sentry alerts) with no possibility of recovery.